globals.go

```diff
@@ -216,6 +220,8 @@ const (
 	PatchVersionRequester       = "patch_request"
 	GithubPRRequester           = "github_pull_request"
 	RepotrackerVersionRequester = "gitter_request"
+	TriggerRequester            = "trigger_request"
+	UserInitiatedRequester      = "user_initiated"
```

Comment:
I don't like that they all end with _request, but... 

Feels weird that user_initiated is different. 

Also user initiated feels like the wrong descriptor for "projectless/freeagent" also I suspect that user-initiated will be one of the key ways that users add new repository types and do cool workflow things, and it'll feel as awkward as "gitter", so I want to try and see if we can sand off some of the rough corners.

Reply:
I renamed it to ad hoc, though if you can think of a description that encompasses all the relevant use cases we should change it to that before it starts being used